src/java.base/share/classes/sun/security/ec/ECDSASignature.java

```diff
@@ -436,7 +442,14 @@ private byte[] signDigestImpl(ECDSAOperations ops, int seedBits,
             random.nextBytes(seedBytes);
             ECDSAOperations.Seed seed = new ECDSAOperations.Seed(seedBytes);
             try {
-                return ops.signDigest(s, digest, seed);
+                // a new allocation for this value, rather than just
+                // returning it, is a trade-off to zero-out the local
+                // value for "s" when necessary
+                byte[] retValue = ops.signDigest(s, digest, seed);
+                if(extraCopyToZero) {
```

Comment:
I believe you could avoid the change above (428-435) by checking `if (!priv instanceof ECPrivateKeyImpl)` here.
Also a style nit with a missing space between if "if" and "("